orc8r/cloud/go/test_utils/unit_test_proto_utils.go

```diff
@@ -35,44 +35,66 @@ func AssertErrorsEqual(t *testing.T, expectedError, actualError string) {
 	expectedErrorParts := strings.Split(expectedError, Separator)
 	// check if actual message contains relevant parts of expected error
 	for _, messagePart := range expectedErrorParts {
-		assert.Contains(t, actualError, messagePart, "actual error does not contain all parts of expected error\nexpected: %v\nactual:   %v", strings.Join(expectedErrorParts, " "), actualError)
+		assert.Contains(t, actualError, messagePart, "actual error does not contain all parts of "+
+			"expected error\nexpected: %v\nactual:   %v", strings.Join(expectedErrorParts, " "), actualError)
 	}
 }
 
-// AssertMapsEqual compares two maps of with proto messages.
-func AssertMapsEqual(t *testing.T, expected, actual map[string]protoV1.Message) {
-	assert.Equal(t, len(expected), len(actual))
-	for key := range actual {
-		AssertMessagesEqual(t, protoV1.MessageV2(expected[key]), protoV1.MessageV2(actual[key]))
+// AssertMapsEqual compares two maps of proto messages.
+func AssertMapsEqual(t *testing.T, expected, actual interface{}) {
+	expValue, actValue := reflect.ValueOf(expected), reflect.ValueOf(actual)
+	if ekind, akind := expValue.Kind(), actValue.Kind(); ekind != akind || ekind != reflect.Map {
+		assert.Equal(t, expected, actual)
+	}
+	assert.Equal(t, expValue.Len(), actValue.Len())
+	for _, key := range expValue.MapKeys() {
+		AssertMessagesEqual(t, expValue.MapIndex(key).Interface(), actValue.MapIndex(key).Interface())
+	}
```

Comment:
What happen to this function if we pass two int?

Reply:
Right, that would result in a panic error because `int` does not include `Len()` as well as `MapsKeys()`. 
I can use the `Kind() != reflect.Map` check before to trigger an assert error.

Reply:
Since this is a test function that wil help to catch bugs, i would rather prefer to add more granularity on the messages
```
	ekind := expValue.Kind()
	akind  := actValue.Kind()
	ekind != reflect.Map 
	if ekind != reflect.Map && akind != reflect.Map {
		assert.Fail(t, fmt.Sprintf("Type error: Both arguments are not Maps \nexpected: %+v\nactual : %+v", ekind, akind))
	}
	else if ekind != reflect.Map {
		assert.Fail(t, fmt.Sprintf("Type error: expected argument is not a a Map \nexpected:  %+v",  ekind))
	}
	esle if akind != reflect.Map {
		assert.Fail(t, fmt.Sprintf("Type error: first argument is not a a Map actual  : %+v",  akind))
	} else...

```

Reply:
@uri200 I modified the test function. Do you have any further comments? Otherwise I would merge the PR.